### ISSUE TITLE
warning with zonal median or sd

### ISSUE THREAD

**Issue by geocaruso** [rspatial/terra#1197]
Hello, I can't get my head around the warning "[rast] 1 out of 3 elements of x are not a SpatRaster" when computing median or st dev for zones. I have searched for this warning and zonal over the web but am still stuck. Sorry if this is obvious and a desired behaviour.

```
r <- terra::rast(ncol=6, nrow=6, xmin=0, xmax=100, ymin=0, ymax=100)
z <- r     # zones
values(z)<-rep(1:3, each=12)
x <- r    # variable
terra::values(x)<-c(seq(1:12),10*seq(1:12),100*seq(1:12))
terra::zonal(x,z,mean)  #OK
terra::zonal(x,z,sum) #OK
terra::zonal(x,z,stats::median) #(use of stats because median is not among the fun in terra::zonal)
terra::zonal(x,z,stats::sd) #(use of stats because sd is not among the fun in terra::zonal)
``` 
The last 2 results are correct but why the Warning "[rast] 1 out of 3 elements of x are not a SpatRaster "?
It seems to hint I have several layers ,of which one is not correct, but I don't.

Best
Geoffrey

**Comment by rhijmans** [comment]
I do not see that message. It is probably because you have loaded some other package that does that. Please try in a clean session.

**Comment by geocaruso** [comment]
Thanks. Good to know already that it is on my side!

Have started afresh and have limited pkg. Still searching for the source of that interaction... 
> (.packages())
[1] "terra"     "stats"     "graphics"  "grDevices" "utils"    
[6] "datasets"  "methods"   "base"

**Comment by rhijmans** [comment]
Oh. Then, perhaps, you are not using the current version of terra?

**Comment by geocaruso** [comment]
Indeed. Solved with terra 1.7.39 ! Thanks!
